libcxx/include/__type_traits/datasizeof.h

```diff
@@ -51,6 +51,7 @@ struct __libcpp_datasizeof {
   // the use as an extension.
   _LIBCPP_DIAGNOSTIC_PUSH
   _LIBCPP_CLANG_DIAGNOSTIC_IGNORED("-Winvalid-offsetof")
+  _LIBCPP_GCC_DIAGNOSTIC_IGNORED("-Winvalid-offsetof")
```

Comment:
Where does this change come from?

Reply:
I noticed from earlier when playing around with `__libcpp_datasizeof` that GCC errors out on the `offsetof` (the CI build failed). It seems that with those diagnostic macros you need to suppress for both Clang and GCC separately because their flags may be different.

Anyway, this should probably a separate issue/PR, so I'll remove it for now.